src/test/java/org/mozilla/zest/test/v1/ZestRequestUnitTest.java

```diff
@@ -44,4 +46,31 @@ public void testTokenReplacement() throws Exception {
 		assertEquals ("test=JKL&user=12JKL34", req2.getData());
 	}
 
+	@Test
+	public void testTokenReplacementInCookies() throws Exception {
+		ZestRequest req = new ZestRequest ();
+
+		req.setUrl(new URL("http://www.example.com/app/{{token1}}"));
+		req.setHeaders("Set-Cookie: test={{token2}}");
+		req.setData("test={{token3}}&user=12{{token3}}34");
+		req.addCookie(new ZestCookie("", "12{{token3}}34", "56{{token2}}78", "", new Date(), true));
+
+		ZestRequest req2 = req.deepCopy();
+
+		ZestVariables tokens = new ZestVariables();
+		tokens.setTokenStart("{{");
+		tokens.setTokenEnd("}}");
+		tokens.addVariable("token", "ABC");
+		tokens.addVariable("token1", "DEFG");
+		tokens.addVariable("token2", "GHI");
+		tokens.addVariable("token3", "JKL");
+		req2.replaceTokens(tokens);
+
+		assertEquals ("http://www.example.com/app/DEFG", req2.getUrl().toString());
+		assertEquals ("Set-Cookie: test=GHI", req2.getHeaders());
+		assertEquals ("test=JKL&user=12JKL34", req2.getData());
+		assertEquals ("12JKL34", req2.getZestCookies().get(0).getName());
```

Comment:
This could be done in the previous method, or, if you really want a different method the other assertions should be removed (no need to duplicate them).